docs: Replicates changes made to Infura tutorials (#1470)
MetaMask/metamask-docs: services/tutorials/ethereum/send-a-transaction/use-web3.js.md

description: Send a transaction using Web3.js.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Use web3.js

In this tutorial, you'll send a regular transaction of 0.001 ETH from one account to another using the Web3 JavaScript library.

## Prerequisites

- A [Web3 project](../../../api/getting-started.md) on Infura
- [Node.js installed](https://nodejs.org/en/download/)
- An Ethereum account

:::info

Use [MetaMask](https://metamask.io) or similar to create an Ethereum account for testing.

:::

## Steps

### 1. Select your network and verify funds

<Tabs>
  <TabItem value="View" label="Sepolia" default>

Verify that your wallet has testnet ETH for the Sepolia network. You may use the [Infura faucet](https://www.infura.io/faucet) to top up. 

</TabItem>
<TabItem value="Retrieve" label="Alternative network" default>

To use an alternative network:

> - Update your [environment file](#4-create-the-env-file) with the alternative network name 

Reviewer: I'm not sure, but should this be numbered as a procedure? I know it does not have to ordered, but maybe this is ordinal?

Create a `.env` file in your project directory to store the project and Ethereum account details.

<Tabs>
  <TabItem value="View" label="Syntax" default>

Replace the following values in the `.env` file:

```bash
ETHEREUM_NETWORK = "<network>"
INFURA_API_KEY = "<API-KEY>"
SIGNER_PRIVATE_KEY = "<PRIVATE-KEY>"
```

> - `<NETWORK>` with "sepolia" or the network you are using.

Reviewer: ```suggestion
- `<NETWORK>` with `"sepolia"` or the network you are using.
```

- `<API-KEY>` with your API key of the web3 project.
- `<PRIVATE-KEY>` with the [private key of your Ethereum account](https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key). A transaction must be signed with the sender's private key. Make sure that you prefix the `SIGNER_PRIVATE_KEY` value with `0x`. The private key you export from MetaMask isn't prefixed with `0x`.

</TabItem>
<TabItem value="Retrieve" label="Example" default>

```text
ETHEREUM_NETWORK = "sepolia"
INFURA_API_KEY = "d23...x...6e"
SIGNER_PRIVATE_KEY = "0x561...x...61df"
```
</TabItem>
</Tabs>

:::danger

Never disclose your private key. Anyone with your private keys can steal the assets controlled by those keys.

:::

### 5. Create `send.js` file

In this example, we'll create a JavaScript file (`send.js`) in the project directory which configures and sends the transaction.

:::info

> Optional: to send test ETH to an account of your choice, update line 20 with your selected account.

Reviewer: ```suggestion
To send test ETH to a specific account, update line 20 with the desired account address.
```

or 

```suggestion
To send test ETH to a specific account, update line 20 with your selected account address.
```
